### PR TITLE
Expose the `run_multiple` Functionality

### LINKED ISSUE #23: Expose the run_multiple functionality
One of the key functions of epiworld is the out of the box capability of running multiple simulations using OpenMP. We need to expose this as it is done in epiworldR. Existing approach:

- Create an object of class saver.
- Implement the method `run_multiple` using the saver object.

More info [here](https://github.com/UofUEpiBio/epiworldR/blob/b4eb34db375daa614da08828b7a549175a6a3570/src/model.cpp#L61-L123) and [here](https://github.com/UofUEpiBio/epiworldR/blob/b4eb34db375daa614da08828b7a549175a6a3570/R/make_saver.R#L4). Notice that R deals with the random name (id) for the saver files.

### PR DESCRIPTION
**Pull request by IsaccBarker**
Fixes #23. There's a bunch of filesystem stuff here, since we're on C++11 and don't have access to the C++17 filesytem APIs. The only thing we should have to worry about here is that the implementation of the Saver constructor matches the [Saver constructor (or equivalent) in EpiworldR](https://github.com/UofUEpiBio/epiworldR/blob/main/R/make_saver.R).

The thing that most needs to be watched out for is what happens if you supply `fn` a file versus a directory; I've done some testing and it appears to be working fine, but there might be some edge cases I didn't test. I would write a unit test to check, but I'm not entirely sure about the intended behaviour of the original R code.

This code will be cleaned up by #24.

### CODE REVIEW

File: tests/test_saver.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        transmission_rate = .1,
+        incubation_days   = 7.0,
+        recovery_rate     = 0.14
+    )
+
+    saver = epiworld.Saver("total_hist", "virus_hist")
+
+    saver.run_multiple(covid19, 100, 4, nthreads=1)
+
+    # TODO: Verify things worked correctly, as is the point of tesing.
```
Comment (IsaccBarker) [inline tests/test_saver.py:18]: *TODO:* Verify things worked correctly, as is the point of testing.

**Review by IsaccBarker — COMMENTED** [review]
I would select the request changes option here, but it won't let me!

File: tests/test_saver.py (hunk 1 above)
Comment (IsaccBarker) [reply tests/test_saver.py:18]: Yeah, getting the results is available (see `get_generation_time`, `get_reproductive_number`, and so on) when iterating over the viruses returned by Epiworld. I've gone ahead and merged the commits with those functions into this PR, but  when using them in a run-multiple context, they only ever seem to report one virus.  For example, the below prints 1 (when we're running doing five runs):

```python
import epiworldpy as epiworld


covid19 = epiworld.ModelSEIR(
    name              = 'covid-19',
    n                 = 10000,
    prevalence        = .01,
    contact_rate      = 2.0,
    transmission_rate = .1,
    incubation_days   = 7.0,
    recovery_rate     = 0.14
)

saver = epiworld.Saver("total_hist", "virus_hist")
saver.run_multiple(covid19, 100, 5, nthreads=1)

generation_time = covid19.get_db().get_generation_time()
viruses = generation_time['viruses']
print(len(np.unique(viruses)))
```

Am I getting the data wrong? I know it can't be a problem with the C++ interface because it's just a direct wrapping of the C++ arrays within Python capsules/arrays.

**Comment by IsaccBarker** [comment]
Blocked by #33.

**Comment by gvegayon** [comment]
Hey @IsaccBarker, how is progress in here?

**Comment by IsaccBarker** [comment]
Solved by #38.
